tokio-util/src/sync/poll_semaphore.rs

```diff
@@ -83,3 +78,34 @@ impl fmt::Debug for PollSemaphore {
             .finish()
     }
 }
+
+struct PollSemaphoreStream {
+    semaphore: Arc<Semaphore>,
+    fut: ReusableBoxFuture<Result<OwnedSemaphorePermit, AcquireError>>,
+}
```

Comment:
You could just merge this with the `PollSemaphore` struct, avoiding duplication of the `semaphore` field.

Reply:
nice simplification